steepshot_io/graph/views.py

```diff
@@ -1,32 +1,48 @@
 import requests
+import logging
 from django.shortcuts import render
 from django.views.generic import View
-from django.conf  import settings
+from django.conf import settings
+from json.decoder import JSONDecodeError
+from requests.exceptions import HTTPError
 
+logger = logging.getLogger(__name__)
 
-class GetPostFee(View):
-    template_name = 'fee_posts.html'
 
-    def get_fee_posts(self, currency='VESTS'):
+class GetPostFee(View):
+    template_name = 'graph.html'
 
-        res = requests.get(settings.REQUESTS_URL_STEEM['post_fee'] + currency).json()
-        res.reverse()
+    def _get_data(self, currency='VESTS', name_key=None, name_url=None, use_fee=True):
         values = []
-        for i in res:
-            values.append([i['day'], i['count_fee']])
+        try:
+            if not use_fee:
+                currency = ''
+            cur_url = settings.REQUESTS_URL[name_url].format(url=settings.STEEM_V1)
+            res = requests.get(cur_url + currency).json()
+            res.reverse()
+            values = []
+            for i in res:
+                values.append([i['day'], i[name_key]])
+        except JSONDecodeError as e:
+            logger.error('Failed to parse json {}'.format(e))
+        except (ConnectionError, HTTPError) as e:
+            logger.error('Failed to connect to {}'.format(e))
+        except Exception as e:
+            logger.error(e)
         return values
 
     def get(self, request, *args, **kwargs):
         if 'currency' in request.GET:
             currency = request.GET['currency'].lower()
-            values = self.get_fee_posts(currency)
+            values = self._get_data(currency=currency, name_key='count_fee', name_url='post_fee')
         else:
-            values = self.get_fee_posts()
-        return render(request, self.template_name, {'values': values})
+            values = self._get_data(name_key='count_fee', name_url='post_fee')
+        name = 'fee in day'
+        return render(request, self.template_name, {'values': values, 'name_1': name})
 
 
 class GetPostsCountMonthly(View):
-    template_name = 'count_posts.html'
+    template_name = 'graph.html'
 
     def group_steem_golos(self, list_steem, list_golos):
```

Comment:
Maybe it's better to make more universal function?

For example this function can merge any count of arrays (but there is no checks to exclude IndexError):
``` python
    def group_steem_golos(self, *args):
        """
            Merge lists by first element of sublist and sum the second

            Example:
            l1 = [('2017-09-08', 10), ('2017-09-09', 12)]
            l2 = [('2017-09-08', 40), ('2017-09-09', 51)]
            result = [('2017-09-08', 50), ('2017-09-09', 63)]
        """
        if args:
            return list(map(lambda x: (x[0][0], sum([y[1] for y in x])), zip(*args)))
        else:
            return []
```

Reply:
![123](https://user-images.githubusercontent.com/23061626/31308340-0c0e5366-ab7e-11e7-8950-78e609e059d1.png)

Reply:
your function do not work

Reply:
It works well, you copied class method and tried to call it like a separate function. You didn't remove 'self' argument from arguments.